src/main/learnocaml_main.ml

```diff
@@ -391,23 +408,24 @@ let main o =
       end
     else Lwt.return_unit
   in
-  let generate o =
+  let generate ?(check_port = true) o =
     if List.mem Build o.commands then
       (let get_app_dir o =
          if not (List.mem Serve o.commands) then
            Lwt.return o.app_dir
-         else if o.server.Server.replace then
-           let app_dir = temp_app_dir o in
-           (if Sys.file_exists app_dir then
+         else if o.server.Server.replace || o.serve_during_build then
```

Comment:
Yes, I hesitated at first, but I believe both commands have a different role:

* `--serve-during-build` is an option that can only be passed to `learn-ocaml build serve` (not `learn-ocaml serve` only)
* `--replace` is an option that can also be passed solely to `learn-ocaml server` (and its behavior, replace the running server process, mostly deals with the server phase)

Let me know if I convinced you! or not.

Reply:
Actually I found another reason why we cannot just add `--serve-during-build` in SERVER OPTIONS:
we don't want to allow `learn-ocaml-server --serve-during-build` (which would be the case given the way the cmdliner options are factor-out between `learn-ocaml serve` and `learn-ocaml-server`).
So I guess we can keep the location of this cmdliner option as is :)

Reply:
Ah I see ; while I agree that code-wise they are pretty different, from a user-perspective, they're both options concerning the server though (as the name implies!) -- and with similar use-cases so it would be best to have them closer to each other.
We could cheat by keeping this organisation in the code, but hardcoding `~docs` to still put it in the SERVER section of the manpage (`COMMON OPTIONS` implicitely means the union of the commands, here we're dealing with the intersection :) )